docs/how_to/variables_and_references.md

```diff
@@ -7,6 +7,157 @@ This guide shows you how to use one of the advanced features of Lumen: Variables
 
 ## Overview
 Variables are one of the most powerful features of Lumen.
-This guide will give you an overview of the different variables and how to reference them in the YAML specification.
+This guide will give you an overview of three different types of variables and how to reference them in the YAML specification.
+
+As a rule of thumb, internal variables are referenced with a starting dollar sign `$`, whereas external references use double curly brackets before and after `{{ }}`.
 
 ## Variables
```

Comment:
I see references explained here and template variables but not a single example of an actual variable definition.

Reply:
Note that's not a great example. I'd probably opt for some widget variable that controls the parameter of a transform.

Reply:
This part is a work-in-progress as I'm missing a working example. It is hard to document it when I'm unsure if it works.

Reply:
I'm aware that this is missing. Did you read my previous comment? 

I'm not able to get the interactive part of the Variables as `Resolving nested variable references currently not supported` so they are just static.

Reply:
Still looking into that but I don't think that blocks you from building an example where a variable controls a transform parameter for instance.